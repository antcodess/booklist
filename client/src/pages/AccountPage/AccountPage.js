import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import BreadCrumbNav from '../../components/BreadCrumbNav/BreadCrumbNav';
import OrdersContainer from '../../components/Orders/OrdersContainer';

import { selectCurrentUser } from '../../redux/user/user.selectors';
import { selectOrderItems } from '../../redux/orders/orders.selectors';
import { fetchOrdersStart } from '../../redux/orders/orders.actions';

import './AccountPage.scss';

const AccountPage = ({ currentUser, fetchOrdersStart, orderItems }) => {
  const { id, displayName } = currentUser;
  useEffect(() => {
    fetchOrdersStart(id);
  }, [fetchOrdersStart, id])
  
  return (
    <React.Fragment>
      <BreadCrumbNav title='Account' /> 
      <section className='account-page'>
        <h2>Account</h2>
        <h3>{`Hello, ${displayName}!`}</h3>
        <span className='edit-profile'>Edit Profile</span>
        <div className='orders-overview'>
          <h4>Orders</h4>
          <div className='orders-seperator'></div>
          <OrdersContainer orderItems={orderItems} />
        </div>
      </section>
    </React.Fragment>
  );
};

const mapStateToProps = createStructuredSelector({
  currentUser: selectCurrentUser,
  orderItems: selectOrderItems
});

const mapDispatchToProps = dispatch => ({
  fetchOrdersStart: currentUserId => dispatch(fetchOrdersStart(currentUserId))
})

export default connect(mapStateToProps, mapDispatchToProps)(AccountPage);