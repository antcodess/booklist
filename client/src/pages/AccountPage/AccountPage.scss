.account-page {
  text-align: left;
  
  h2 {
    margin-top: 20px;
    font-size: 2rem;
  }
  h3 {
    margin: 20px 0 10px;
    font-size: 1.2rem;
  }
  
  .edit-profile {
    font-size: 0.94rem;
  }
  
  .orders-overview {
    margin-top: 30px;

    h4 {
      margin: 20px 0 10px;
      font-size: 1.5rem;
    }

    .orders-seperator {
      border-bottom: 2px solid #ededed;
      margin-bottom: 20px;
      width: 76px;
    }

    .orders-empty-message {
      color: #6A6A6A;

      .shop-link {
        color: #6A6A6A;
        font-family: "Hurme Geometric Sans 3 W03 Bold";

        &:hover {
          color: #0080DE;
        }
      }
    }
  }
}