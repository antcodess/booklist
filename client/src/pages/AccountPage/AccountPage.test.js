import React from 'react';
import { shallow } from 'enzyme';
import { AccountPage } from './AccountPage';
import { mapDispatchToProps } from './AccountPage';

describe('AccountPage Component', () => {
  let wrapper;
  let mockCurrentUser;
  let mockFetchOrdersStart;
  let mockOrderItems;
  let mockHistory;
  let dummyFunction = jest.fn();

  const mockPayload = {
    type: 'FETCH_ORDERS_START',
    payload: 1
  };
  
  beforeEach(() => {

    mockCurrentUser = { id: 1, displayName: 'Sample User'};
    mockFetchOrdersStart = jest.fn();
    mockOrderItems = [{id: 1}, {id: 2}, {id: 3}];
    mockHistory = { push: jest.fn() };

    const mockProps = {
      currentUser: mockCurrentUser, 
      fetchOrdersStart: mockFetchOrdersStart, 
      orderItems: mockOrderItems,
      history: mockHistory
    };
    
    wrapper = shallow(<AccountPage { ...mockProps } />);
  });

  it('should render AccountPage Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push when clicked for edit-profile', () => {
    wrapper.find('.edit-profile').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should dispatch fetchOrdersStart action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchOrdersStart(1);
    expect(dispatch.mock.calls[0][0]).toEqual(mockPayload);
  });
});
