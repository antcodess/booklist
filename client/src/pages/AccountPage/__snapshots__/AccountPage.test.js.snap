// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountPage Component should render AccountPage Component 1`] = `ShallowWrapper {}`;
